cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -292,10 +292,15 @@ func getControllerManagerCommand(cfg *kubeadmapi.InitConfiguration, k8sVersion *
 		"cluster-signing-key-file":         filepath.Join(cfg.CertificatesDir, kubeadmconstants.CAKeyName),
 		"use-service-account-credentials":  "true",
 		"controllers":                      "*,bootstrapsigner,tokencleaner",
-		"authentication-kubeconfig":        filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName),
-		"authorization-kubeconfig":         filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName),
-		"client-ca-file":                   filepath.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
-		"requestheader-client-ca-file":     filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyCACertName),
+	}
+
+	//add the extra arguments for v1.12+
+	miniVersion, _ := version.ParseSemantic("v1.12.0")
```

Comment:
s/miniVersion/v112args

Reply:
In current form, this will not allow to deploy Beta/RC versions of 1.12 to be deployed.
Check when those files are added and use something like 'v1.12.0-beta.0'.
Plus, might be instead of hardcoding constant here, it will be better to declare it in `constants/constants.go`